### ISSUE TITLE
[bug] FreeBSD compilation fails on scrypt.h

### ISSUE THREAD

**Issue by patricklodder** [dogecoin/dogecoin#2475]
# Bug Report

**Current behavior:**

Compiling Dogecoin Core (without wallet) on FreeBSD 11.4 fails with the following error:

```
  CXX      primitives/libdogecoin_common_a-pureheader.o 

In file included from primitives/pureheader.cpp:9:                              
./crypto/scrypt.h:30:24: error: redefinition of 'le32dec'                       
static inline uint32_t le32dec(const void *pp)                                  
                       ^                                                        
/usr/include/sys/endian.h:132:1: note: previous definition is here              
le32dec(const void *pp)                                                         
^                                                                               
In file included from primitives/pureheader.cpp:9:                              
./crypto/scrypt.h:37:20: error: redefinition of 'le32enc'                       
static inline void le32enc(void *pp, uint32_t x)                                
                   ^                                                            
/usr/include/sys/endian.h:186:1: note: previous definition is here              
le32enc(void *pp, uint32_t u)                                                   
^                                                                               
2 errors generated.                   
```

**Steps to reproduce:**

On a clean FreeBSD 11.4 install, do:

```bash
pkg install autoconf automake libtool pkgconf
pkg install boost-libs openssl libevent
pkg install gmake
pkg install git
git clone https://github.com/dogecoin/dogecoin.git
cd dogecoin
./autogen.sh 
./configure --disable-wallet
gmake
```

**Dogecoin Core version**: 1.14.4

**Machine specs**
- OS: FreeBSD 11.4
- CPU: QEMU Virtual CPU version 2.5+ (amd64)
- RAM: 2048MB

**Comment by chromatic** [comment]
[Litecoin appears to have fixed that with a define](https://github.com/litecoin-project/litecoin/commit/3105ef37d83b84d9fdf248989be396bcecc11155).

Similar fix in [PivX cryptocurrency project](https://github.com/Fuzzbawls/PIVX/commit/1433cb63fe2c5da2ea29a40db421ef9aef318e07).

**Comment by patricklodder** [comment]
This has been fixed already in 1.21-dev and is only a problem for 1.14.x

**Comment by ReverseControl** [comment]
Who is taking this one up?

**Comment by ReverseControl** [comment]
@patricklodder Are we fixing this for 1.14.4?

**Comment by patricklodder** [comment]
> Are we fixing this for 1.14.4?

No. 1.14.5 is fine.

**Comment by Saint111188** [comment]
> # Bug Report
> 
> 
> 
> **Current behavior:**
> 
> 
> 
> Compiling Dogecoin Core (without wallet) on FreeBSD 11.4 fails with the following error:
> 
> 
> 
> ```
> 
>   CXX      primitives/libdogecoin_common_a-pureheader.o 
> 
> 
> 
> In file included from primitives/pureheader.cpp:9:                              
> 
> ./crypto/scrypt.h:30:24: error: redefinition of 'le32dec'                       
> 
> static inline uint32_t le32dec(const void *pp)                                  
> 
>                        ^                                                        
> 
> /usr/include/sys/endian.h:132:1: note: previous definition is here              
> 
> le32dec(const void *pp)                                                         
> 
> ^                                                                               
> 
> In file included from primitives/pureheader.cpp:9:                              
> 
> ./crypto/scrypt.h:37:20: error: redefinition of 'le32enc'                       
> 
> static inline void le32enc(void *pp, uint32_t x)                                
> 
>                    ^                                                            
> 
> /usr/include/sys/endian.h:186:1: note: previous definition is here              
> 
> le32enc(void *pp, uint32_t u)                                                   
> 
> ^                                                                               
> 
> 2 errors generated.                   
> 
> ```
> 
> 
> 
> **Steps to reproduce:**
> 
> 
> 
> On a clean FreeBSD 11.4 install, do:
> 
> 
> 
> ```bash
> 
> pkg install autoconf automake libtool pkgconf
> 
> pkg install boost-libs openssl libevent
> 
> pkg install gmake
> 
> pkg install git
> 
> git clone https://github.com/dogecoin/dogecoin.git
> 
> cd dogecoin
> 
> ./autogen.sh 
> 
> ./configure --disable-wallet
> 
> gmake
> 
> ```
> 
> 
> 
> **Dogecoin Core version**: 1.14.4
> 
> 
> 
> **Machine specs**
> 
> - OS: FreeBSD 11.4
> 
> - CPU: QEMU Virtual CPU version 2.5+ (amd64)
> 
> - RAM: 2048MB
> 
>

**Comment by ReverseControl** [comment]
@Saint111188 The fix for this one is straightforward, see @chromatic  comment, https://github.com/dogecoin/dogecoin/issues/2475#issuecomment-902902749, Could you implement it and push it?

**Comment by patricklodder** [comment]
Fixed w/ #2481
